crypto/cmp/cmp_http.c

```diff
@@ -0,0 +1,506 @@
+/*
+ * Copyright OpenSSL 2007-2018
+ * Copyright Nokia 2007-2018
+ * Copyright Siemens AG 2015-2018
+ *
+ * Contents licensed under the terms of the OpenSSL license
+ * See https://www.openssl.org/source/license.html for details
+ *
+ * SPDX-License-Identifier: OpenSSL
+ *
+ * CMP implementation by Martin Peylo, Miikka Viljanen, and David von Oheimb.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+#include "e_os.h"
+#include <openssl/asn1.h>
+#include <openssl/asn1t.h>
+#include <openssl/cmp.h>
+#include <openssl/ocsp.h>
+#include <openssl/ssl.h>
```

Comment:
Oops, when I removed dependencies of `crypto/cmp/` on `ssl.h` from `cmp_http.c` I forgot this case, which is just for improving error diagnostics. I just moved it out to `apps/cmp.c`
(while exporting `(OSSL_)CMP_add_error_txt()` and friends because `OSSL_CMP_add_error_data()` is now used in `apps/cmp.c`).